homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,428 @@
+                )
+                self._brightness = None
+        # pylint: disable=broad-except
+        except Exception as ex:
+            _LOGGER.error(ex)
+            self._state = None
+
+    def update_temperature(self):
+        """Update the temperature."""
+        colortemp = self._light.state.get_colortemp()
+        if colortemp is None or colortemp == 0:
+            self._temperature = None
+            return
+        try:
+            _LOGGER.debug(
+                "[wizlight %s] kelvin from the bulb: %s", self._light.ip, colortemp
+            )
+            temperature = color_utils.color_temperature_kelvin_to_mired(colortemp)
+            self._temperature = temperature
+
+        # pylint: disable=broad-except
+        except Exception:
+            _LOGGER.error("Cannot evaluate temperature", exc_info=True)
+            self._temperature = None
+
+    def update_color(self):
+        """Update the hs color."""
+        colortemp = self._light.state.get_colortemp()
+        if colortemp is not None and colortemp != 0:
+            self._hscolor = None
+            return
+        if self._light.state.get_rgb() is None:
+            return
+        try:
+            rgb = self._light.state.get_rgb()
+            if rgb[0] is None:
+                # this is the case if the temperature was changed
+                # do nothing until the RGB color was changed
+                return
+            warmwhite = self._light.state.get_warm_white()
+            if warmwhite is None:
+                return
+
+            self._hscolor = self._light.convertHSfromRGBCW(rgb, warmwhite)
+
+        # pylint: disable=broad-except
+        except Exception:
+            _LOGGER.error("Cannot evaluate color", exc_info=True)
+            self._hscolor = None
+
+    def update_effect(self):
+        """Update the bulb scene."""
+        self._effect = self._light.state.get_scene()
+
+    async def get_bulb_type(self):
+        """Get the bulb type."""
+        if self._bulbtype is not None:
+            return self._bulbtype
+        try:
+            self._bulbtype = await self._light.get_bulbtype()
+            _LOGGER.info(
+                "[wizlight %s] Initiate the WiZ bulb as %s",
+                self._light.ip,
+                self._bulbtype.name,
+            )
+        except WizLightTimeOutError:
+            _LOGGER.debug(
+                "[wizlight %s] Bulbtype update failed - Timeout", self._light.ip
+            )
+
+    async def update_scene_list(self):
+        """Update the scene list."""
+        self._scenes = await self._light.getSupportedScenes()
+
+    async def get_mac(self):
+        """Get the mac from the bulb."""
+        try:
+            self._mac = await self._light.getMac()
+        except WizLightTimeOutError:
+            _LOGGER.debug("[wizlight %s] Mac update failed - Timeout", self._light.ip)
+
+    def featuremap(self):
+        """Map the features from WizLight Class."""
+        features = 0
```

Comment:
```suggestion
        if not self._bulbtype:
            return SUPPORT_FEATURES_RGB
        features = 0
```

Then remove the try and outdent below